src/xpath.rs

```diff
@@ -251,3 +251,16 @@ impl fmt::Display for Object {
     }
   }
 }
+
+/// Calls the binding to http://xmlsoft.org/html/libxml-xpath.html#xmlXPathCompile and return true if 
+/// a non-null pointer is returned. The idea is to use this to validate an xpath independent of context.
+/// Tests describing what this validates in tests/xpath_tests.rs
+pub fn xml_xpath_compiles(xpath: &str) -> bool {
+  let c_xpath = CString::new(xpath).unwrap();
+    let xml_xpath_comp_expr_ptr = unsafe { xmlXPathCompile(c_xpath.as_bytes().as_ptr()) };
+    let result = !xml_xpath_comp_expr_ptr.is_null();
+    if result {
+      unsafe { libc::free(xml_xpath_comp_expr_ptr as *mut c_void); }
+    }
+    result
```

Comment:
I think a slightly more idiomatic rust way of stating the ending here would be:

```rust
if xml_xpath_comp_expr_ptr.is_null() {
  false
} else {
  unsafe { libc::free(xml_xpath_comp_expr_ptr as *mut c_void); }
  true
}
```

Reply:
I agree this is "rustier" and generally easier to read! As far as the function name, I chose it to match the c function. You are correct that the semantic reason I have for "compiling" the xpath is to check that it is well formed, but I didn't want to make that assertion without confirming it. If you can confirm that a non null return from `unsafe { xmlXPathCompile(c_xpath.as_bytes().as_ptr()) };` guarantees a well formed xpath, then I also agree the function name should be changed.

Reply:
I will change the function name in a separate PR and merge there, since I would like to include the functionality, but we don't have to agree on my naming reasoning.

The way I see it, it is not for me to *guarantee* the XPath is well-formed, but for cases where it isn't - but does compile - for users to report bugs to the libxml2 team. The intention of a compiled XPath is that the libxml2 engine has figured out some utility for employing it.